cmake/coreneuron/MakefileBuildOptions.cmake

```diff
@@ -71,20 +72,34 @@ function(coreneuron_process_target target)
       endif()
       get_target_property(target_imported ${target} IMPORTED)
       if(target_imported)
-        # In this case we can extract the full path to the library
-        get_target_property(target_location ${target} LOCATION)
-        coreneuron_process_library_path(${target_location})
+        if(NOT (${target_type} STREQUAL "INTERFACE_LIBRARY"))
+          # In this case we can extract the full path to the library
+          get_target_property(target_location ${target} LOCATION)
+          if(target_location)
+            coreneuron_process_library_path(${target_location})
+          endif()
+        endif()
       else()
         # This is probably another of our libraries, like -lcoreneuron-cuda. We might need to add -L
         # and an RPATH later.
         set_property(GLOBAL APPEND_STRING PROPERTY CORENRN_LIB_LINK_DEP_FLAGS " -l${target}")
       endif()
     endif()
-    get_target_property(target_libraries ${target} LINK_LIBRARIES)
-    if(target_libraries)
-      foreach(child_target ${target_libraries})
-        coreneuron_process_target(${child_target})
-      endforeach()
+    get_target_property(target_flags ${target} INTERFACE_COMPILE_OPTIONS)
+    if(target_flags)
+      if(target_flags MATCHES "\\$<")
+        message(WARNING "Skipping compile options ${target_flags} from target ${target}")
+      else()
+        set_property(GLOBAL APPEND_STRING PROPERTY CORENRN_EXTRA_COMPILE_FLAGS " ${target_flags}")
+      endif()
+    endif()
+    if(NOT (${target_type} STREQUAL "INTERFACE_LIBRARY"))
```

Comment:
Why exclude interface libraries? They're useful to collect a bunch of dependencies into a single target, i.e. an interface library.

Reply:
I had some CMake failures with it here. I'll see if dropping the OpenMP target linkage makes it moot.